From 10e2f3fc7eef6da4d741f7617e80d028257d9884 Mon Sep 17 00:00:00 2001
From: Laszlo Ersek <lersek@redhat.com>
Date: Wed, 25 May 2022 13:06:01 +0200
Subject: [PATCH] update common submodule

Shortlog for 48527b8768d7..f8de5508fe75:

Laszlo Ersek (1):
      mlcustomize: refresh generated files

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=2089748
Signed-off-by: Laszlo Ersek <lersek@redhat.com>
(cherry picked from commit 273de755dfe7eb0f1e81dc62463c125e8bed0cff)
---
 common | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Submodule common 48527b876..f8de5508f:
diff --git a/common/mlcustomize/customize-options.pod b/common/mlcustomize/customize-options.pod
index a83c80a..8aafacd 100644
--- a/common/mlcustomize/customize-options.pod
+++ b/common/mlcustomize/customize-options.pod
@@ -310,6 +310,10 @@ It cannot delete directories, only regular files.
 
 =back
 
+=item B<--selinux-relabel>
+
+This is a compatibility option that does nothing.
+
 =item B<--sm-attach> SELECTOR
 
 Attach to a pool using C<subscription-manager>.
diff --git a/common/mlcustomize/customize-synopsis.pod b/common/mlcustomize/customize-synopsis.pod
index 2520853..9e2c4b2 100644
--- a/common/mlcustomize/customize-synopsis.pod
+++ b/common/mlcustomize/customize-synopsis.pod
@@ -13,4 +13,4 @@
     [--uninstall PKG,PKG..] [--update] [--upload FILE:DEST]
     [--write FILE:CONTENT] [--no-logfile]
     [--password-crypto md5|sha256|sha512] [--no-selinux-relabel]
-    [--sm-credentials SELECTOR]
+    [--selinux-relabel] [--sm-credentials SELECTOR]
diff --git a/common/mlcustomize/customize_cmdline.ml b/common/mlcustomize/customize_cmdline.ml
index 5d404e8..a17bed4 100644
--- a/common/mlcustomize/customize_cmdline.ml
+++ b/common/mlcustomize/customize_cmdline.ml
@@ -111,6 +111,8 @@ and flags = {
       (* --password-crypto md5|sha256|sha512 *)
   no_selinux_relabel : bool;
       (* --no-selinux-relabel *)
+  selinux_relabel_ignored : bool;
+      (* --selinux-relabel *)
   sm_credentials : Subscription_manager.sm_credentials option;
       (* --sm-credentials SELECTOR *)
 }
@@ -122,6 +124,7 @@ let rec argspec () =
   let scrub_logfile = ref false in
   let password_crypto = ref None in
   let no_selinux_relabel = ref false in
+  let selinux_relabel_ignored = ref false in
   let sm_credentials = ref None in
 
   let rec get_ops () = {
@@ -132,6 +135,7 @@ let rec argspec () =
     scrub_logfile = !scrub_logfile;
     password_crypto = !password_crypto;
     no_selinux_relabel = !no_selinux_relabel;
+    selinux_relabel_ignored = !selinux_relabel_ignored;
     sm_credentials = !sm_credentials;
   }
   in
@@ -464,6 +468,12 @@ let rec argspec () =
       s_"Do not relabel files with correct SELinux labels"
     ),
     None, "Do not attempt to correct the SELinux labels of files in the guest.\n\nIn such guests that support SELinux, customization automatically\nrelabels files so that they have the correct SELinux label.  (The\nrelabeling is performed immediately, but if the operation fails,\ncustomization will instead touch F</.autorelabel> on the image to\nschedule a relabel operation for the next time the image boots.)  This\noption disables the automatic relabeling.\n\nThe option is a no-op for guests that do not support SELinux.";
+    (
+      [ L"selinux-relabel" ],
+      Getopt.Set selinux_relabel_ignored,
+      s_"Compatibility option doing nothing"
+    ),
+    None, "This is a compatibility option that does nothing.";
     (
       [ L"sm-credentials" ],
       Getopt.String (
diff --git a/common/mlcustomize/customize_cmdline.mli b/common/mlcustomize/customize_cmdline.mli
index 7ee882a..7d14e78 100644
--- a/common/mlcustomize/customize_cmdline.mli
+++ b/common/mlcustomize/customize_cmdline.mli
@@ -103,6 +103,8 @@ and flags = {
       (* --password-crypto md5|sha256|sha512 *)
   no_selinux_relabel : bool;
       (* --no-selinux-relabel *)
+  selinux_relabel_ignored : bool;
+      (* --selinux-relabel *)
   sm_credentials : Subscription_manager.sm_credentials option;
       (* --sm-credentials SELECTOR *)
 }
-- 
2.31.1

